### PR TITLE
Delete confirm update

### PR DESCRIPTION
**Pull request by TheSecEng**
**In reference to:** #39 

## Purpose
In the current implementation, the confirm dialog presents all files into a single object on the quick look panel. This not only extends the height of each item, it makes it hard to visibly see the cancel button.

### Changes
With the following changes, each file is broken into its own item on the quick look panel into its own row. It also adds the Confirm and Cancel options at the top of the list.

Confirm-Delete-Dialog

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,6 @@ site/
 test.py
 __pycache__/
 Thumbs.db
+.DS_STORE
+*sublime-project
```
Comment (math2001) [inline .gitignore]: I'll leave that in the `.gitignore` for other people, but dealing with `.sublime-project` really annoyed me. I found [`randy3k/ProjectManager`](https://github.com/randy3k/ProjectManager), and now it's THE plugin I would install if I could only install one plugin (FileManager doesn't count as a plugin of course, it's native :smile:). I really recommend you use it (just rebind `ctrl+alt+p` to his command, and you'll never think about `.sublime-projects` again)

Comment (TheSecEng) [reply .gitignore]: I’ll check out that plugin, I feel like the project functions are a bit lacking. So maybe this is a good solution for me.

**Comment by math2001** [comment]
Oh that's a good idea! I really like it!! :hearts: 

I was wondering what it did when I selected just one file. I quickly looked at the code, and I thought that it would delete just that file. I took a bit of time and I realized that it removes just that item from the "to delete" list. I think your option is better.

But it needs to be clear, we can't live the user guessing. How about adding a message in the text under the `Cancel` thing? Maybe it could be something like `Cancel the deletion of x items - select one of the item below remove it from the list`. I don't really like the wording here, don't hesitate to use your own/propose some here if you're not sure.

Btw, we can delete files *and* folders, so we need to pay attention to that in the messages and in the code.

Also, all the code should be formatted using `black`. The current packages for black on sublime are either bloated (sublack) or less bloated but don't use the `blackd` server. I wrote a quick light plugin that just works out of the box, and is fast, but it's not really clean. I might fix it up and push it to GitHub (though I probably won't publish it to Package Control, because sublack already does everything my package will ever do). 

If you don't want to install those plugins, just run `black .` before you commit, it'll fix up your code for you.

Damn, I just merged #59, so this is going to conflict if you want to make some changes. Let me know if you have any trouble with that.

File: FMcommands/delete.py [hunk 2]
```diff
@@ -25,16 +39,29 @@ def run(self, paths=None, *args, **kwargs):
...
+                    self.paths) > 1 else "")
+            )
+            paths_to_display = [
+                ["Confirm", confirm_text],
+                ["Cancel", cancel_text]
+            ]
+
+            for path in paths:
+                paths_to_display.append(
+                    [path.split(os.path.sep)[-1], path])
```
Comment (math2001) [inline FMcommands/delete.py]: Honestly, that's fine. But it reads a bit better if you use `os.path.basename` instead of `path.split(os.path.sep)[-1]`.

Comment (TheSecEng) [reply FMcommands/delete.py]: Didn't even realize that was a thing! Done 👍

File: FMcommands/delete.py [hunk 3]
```diff
@@ -25,16 +39,29 @@ def run(self, paths=None, *args, **kwargs):
...
+                    len(self.paths)) if len(self.paths) > 1 else ""),
+                ("s" if len(
+                    self.paths) > 1 else "")
+            )
+            cancel_text = "Cancel deletion of {0}file{1}".format(
+                ("{0} ".format(
+                    len(self.paths)) if len(self.paths) > 1 else ""),
+                ("s" if len(
+                    self.paths) > 1 else "")
+            )
```
[the remark this answers is not in the document]
Comment (TheSecEng) [reply FMcommands/delete.py]: Implemented these changes and adjusting wording. 

Thanks for the feedback!

**Comment by TheSecEng** [comment]
> Damn, I just merged #59, so this is going to conflict if you want to make some changes. Let me know if you have any trouble with that.
> 
> Just to be clear, two more things that can need to be done:
> 
> * [ ]  format using black
> * [ ]  improve the cancel message

Done and done. 

* [X]  format using black
* [X]  improve the cancel message

**Comment by TheSecEng** [comment]
Just to show the change of wording


image

**Comment by math2001** [comment]
:tada: Thanks a bunch @TheSecEng, great idea!

**Comment by TheSecEng** [comment]
@math2001 You are right, FileManager is amazing! Especially when combine with Deanishe's Alfred workflow for sublime projects.


image

**Pull request merged**